packages/@dcl/ecs/src/serialization/crdt/putComponent.ts

```diff
@@ -14,17 +13,12 @@ export namespace PutComponentOperation {
    * Call this function for an optimal writing data passing the ByteBuffer
    *  already allocated
    */
-  export function write(
-    entity: Entity,
-    timestamp: number,
-    componentDefinition: ComponentDefinition<unknown>,
-    buf: ByteBuffer
-  ) {
+  export function write(entity: Entity, timestamp: number, componentId: number, data: Uint8Array, buf: ByteBuffer) {
     // reserve the beginning
     const startMessageOffset = buf.incrementWriteOffset(CRDT_MESSAGE_HEADER_LENGTH + MESSAGE_HEADER_LENGTH)
 
     // write body
-    componentDefinition.writeToByteBuffer(entity, buf)
```

Comment:
This prevented to allocate a new Uint8Array to serialize the component and directly serializing in `buf`

Reply:
We can take the note and optimize in the future